test/gpu/amdgpu.jl

```diff
@@ -6,10 +6,18 @@ using LinearOperators, AMDGPU
   B = ROCArray(rand(Float32, 10, 10))
   C = ROCArray(rand(Float32, 20, 20))
   M = BlockDiagonalOperator(A, B, C)
+  v = ROCArray(rand(5))
+
 
   v = ROCArray(rand(Float32, 35))
   y = M * v
   @test y isa ROCArray{Float32}
 
```

Comment:
if you move the assignment to v of `ROCArray(rand(5))` here, it should work as expected.